src/Compilers/Core/Portable/Syntax/CollectionBuilderAttribute.cs

```diff
@@ -7,7 +7,7 @@
 namespace System.Runtime.CompilerServices
 {
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface)]
-    public sealed class CollectionBuilderAttribute : Attribute
+    internal sealed class CollectionBuilderAttribute : Attribute
```

Comment:
Prior to this PR, was it accidental that this was declared public? Or has something changed so that only now we want it to be internal?

Reply:
prior to this pr this was only defined in the test layer.  now it's pulled down into the actual compiler layer effectively as a polyfill.  so we're saying:

prior to .net8, we add our own attribute here, so that our compiler can see this when people target this api.
after .net8, we pick up the actual real attribute.